src/md.js

```diff
@@ -0,0 +1,37 @@
+const fs = require('fs');
+const path = require('path');
+
+const mdLinks = (pathDocument) => new Promise((resolve, err) => {
+
+  const toAbsolutePath = path.resolve(pathDocument);
+  const docExtension = path.extname(toAbsolutePath);
+  /* const linksRegExp = /\[((.+?))\]\((http|https).+?\)/g;
+  const hrefRegExp = /\((http|https).+?\)/g; */
+  const textLinksRegExp = /\[.+?\]/g;
+
+  if (docExtension === '.md') {
+
+    const docMdToString = fs.readFileSync(toAbsolutePath).toString();
+    const linksMatchInMd = docMdToString.match(linksRegExp);
+ 
+    const arrayLinksMd = [];
+
+    for (let i in linksMatchInMd) {
```

Comment:
esta raro el for, solo quiero recordarte su estructura for(let i=algo, i<=n, i++) lo conforma la asignación, después la evaluación y por ultimo la reacción en caso de que se cumpla.
```suggestion
    for (let i in linksMatchInMd) {
```